modules/engine/src/listeners.ts

```diff
@@ -650,11 +672,56 @@ async function handleConditionalTransferCreation(
   };
   evts[EngineEvents.CONDITIONAL_TRANSFER_CREATED].post(payload);
 
-  // If we should not route the transfer, do nothing
-  if (!routingId || transfer.meta?.routingId !== routingId) {
-    logger.warn({ transferId, routingId, meta: transfer.meta }, "Cannot route transfer");
+  // TRANSFER_ROUTING_COMPLETE event
+
+  // if transfer is created and we are the end recipient, emit event now and end
+  if (
+    transfer.responderIdentifier === signer.publicIdentifier &&
+    (transfer.meta as RouterSchemas.RouterMeta)?.path[0]?.recipient === signer.publicIdentifier
+  ) {
+    evts.CONDITIONAL_TRANSFER_ROUTING_COMPLETE.post({
+      publicIdentifier: signer.publicIdentifier,
+      initiatorIdentifier: transfer.initiatorIdentifier,
+      responderIdentifier: transfer.responderIdentifier,
+      routingId,
+      meta: transfer.meta,
+    });
     return;
   }
+
+  // if we are the exit router, also send out-of-protocol NATS message to the sender
+  // of the linked transfer so they know the path is complete (and can forward to their sender and so on)
+  if (
```

Comment:
nit: alternative to return above, you could just make this an `else if`, that way if there's any more NATS messages we need to send here based on other conditions in the future, no room for mistake

but that's nit. otherwise lgtm 👍